### ISSUE TITLE
Long data type for Timestamps in Timeseries chart

### ISSUE THREAD

**Issue by utkarshns** [PhilJay/MPAndroidChart#2203]
Please add `long` datatype support for unix timestamps in the timeseries LineChart.
The current `float` datatype isn't precisely able to store timestamps which are near to each other, as a result the end bits are truncated and the nearby values end up being overwritten leading to incorrect plots.

**Comment by utkarshns** [comment]
Any updates?
@PhilJay

**Comment by utkarshns** [comment]
@PhilJay
Anything?

**Comment by zkdzegede** [comment]
For a temporary fix.

I just record the first timestamp myself, and then use the diff between the first timestamp and subsequent ones, so that my xAxis starts from 0 and increases by 1000 for each second.

**Comment by utkarshns** [comment]
@zkdzegede 
Even I am using that sort of technique but it adds to extra processing, it would be great to just have a larger datatype.

@PhilJay anything?

**Comment by xelnaga** [comment]
Same problem.  This is a big limitation.

**Comment by zkdzegede** [comment]
@utkarshns  

I agree it would be nice, but a double could hurt performance so I could see why they chose float  

:(

**Comment by LunarWatcher** [comment]
I know I'm rather late on this issue, but there is a way Double and/or Long could be added.

While they apparently hurt performance, anyone who wants to use it should be allowed to. Maintaining Float as the "main" data type but adding support for Double and Long would allow those that need fast charts to use Float and those who need accurate charts and don't care about performance to use Double and Long. A warning could be added to warn anyone who wants to use it that it does hurt performance.

This way everyone is happy, as it supports multiple data types while still having the option of taking a fast chart. 

I'm basically saying keep float, but add support for double and long. Keep float as the "main" data type and add different classes that have different names for Double and Long. The existing code (if this is implemented) would still be Float based unless it is rewritten with the class names for Double or Long

**Comment by nisiode** [comment]
+1

**Comment by eurbon** [comment]
It's 2019 and we still dont have ability to use longs for timestamps?

**Comment by utkarshns** [comment]
True, you can however have a workaround. Please look at earlier messages
for the same.

**Comment by khushbushah2302** [comment]
Hello, any update on this ? 
I have the same issue and I am looking for a solution to plot a lot of points where X is unix timestamp.
